### PR TITLE
Migrate to new commands API

### PR DESCRIPTION
**Pull request by pkukielka**

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt [hunk 1]
```diff
@@ -102,7 +102,7 @@ private constructor(
 
       val process =
           processBuilder
-              .redirectErrorStream(false)
+              .redirectErrorStream(true)
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt]: can you say smth about this change?

File: src/main/kotlin/com/sourcegraph/cody/commands/ui/CommandsTabPanel.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    commands.forEach { (command, text) -> addButton(command, text) }
+    CommandsContextMenu.addCommandsToCodyContextMenu(project, commands) {
+      executeCommandWithContext(it)
+    }
+  }
+
+  fun enableAllButtons() = switchAllButtons(isEnabled = true, tooltip = null)
+
+  fun disableAllButtons() =
+      switchAllButtons(isEnabled = false, tooltip = "Message generation in progress...")
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/commands/ui/CommandsTabPanel.kt]: nit: bundle the message

File: src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt [hunk 3]
```diff
@@ -497,7 +369,7 @@ class CodyToolWindowContent(private val project: Project) : UpdatableChat {
     ApplicationManager.getApplication().executeOnPooledThread {
       val chat = Chat()
       try {
-        chat.sendMessageViaAgent(project, humanMessage, recipeId, this, inProgressChat)
+        chat.sendMessageViaAgent(project, humanMessage, commandId, this, inProgressChat)
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt:350]: nit: here it's `commandId` but in `sendMessageViaAgnet` it's just `command`

File: src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt [hunk 4]
```diff
@@ -52,7 +47,8 @@ class CodyToolWindowContent(private val project: Project) : UpdatableChat {
   private var inProgressChat = CancellationToken()
   private val stopGeneratingButton =
       JButton("Stop generating", IconUtil.desaturate(AllIcons.Actions.Suspend))
-  private val recipesPanel: JBPanelWithEmptyText
+  private val commandsPanel: CommandsTabPanel =
+      CommandsTabPanel(project) { cmd: String -> sendMessage(project, message = null, cmd) }
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt]: I wonder... does it make sense to have a type for commands?

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt (hunk 1 above)
Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt]: Good catch, it's not needed, just for my local testing.

File: src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt (hunk 4 above)
Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt]: It does and I will add it, it was just a quick WiP PR to check if it works.

File: build.gradle.kts [hunk 5]
```diff
@@ -198,7 +198,7 @@ tasks {
     return destinationDir
   }
 
-  val codyCommit = "268f400ba28f332432e5282c0624d39e52af0848"
```
Comment (mkondratek) [inline build.gradle.kts:201]: ❗ CODY COMMIT UPDATED

Comment (mkondratek) [reply build.gradle.kts:201]: in particular:
- it brings the fix to the selection

**Review by mkondratek — APPROVED** [review]

**Pull request merged**